### ISSUE TITLE
Folder structure gets messed up on drag&drop upload

### ISSUE THREAD

**Issue by SanokKule** [rejetto/hfs#719]
Folder structure gets messed up on drag&drop upload
[test-folder.zip](https://github.com/user-attachments/files/16789254/test-folder.zip)
```
test-folder:
  file1.txt
  subfolder1:
    file2.txt
    subsubfolder:
      file4.txt
  subfolder2:
    File4.txt
```
becomes
```
test-folder:
  subfolder1:
    subsubfolder:
      test-folder:
        subfolder1:
          subsubfolder:
            file4.txt
    test-folder:
      subfolder1:
        file2.txt
  subfolder2:
    test-folder:
      subfolder2:
        file3.txt
  test-folder: <- this folder has uploader listed
    file1.txt
```

**Environment (please complete the following information):**
 - OS: Windows 11 23H2
 - HFS 0.54.0-alpha1
 - Browser: Firefox 129

**Additional context**
This only happens with drag&drop and only on firefox(?), if uploaded with the upload button structure is fine.
Also uploader is not shown properly in the frontend for either case, even though it's written on all files/folders.

**Comment by rejetto** [comment]
thank you for reporting!
i got to reproduce it, working on it

**Comment by rejetto** [comment]
[hfs-linux-x64-0.54.0-alpha1.2.zip](https://github.com/user-attachments/files/17085696/hfs-linux-x64-0.54.0-alpha1.2.zip)
[hfs-windows-x64-0.54.0-alpha1.2.zip](https://github.com/user-attachments/files/17085697/hfs-windows-x64-0.54.0-alpha1.2.zip)

**Comment by rejetto** [comment]
fixed in [hfs-windows-x64-0.54.0-alpha1.3.zip](https://github.com/user-attachments/files/17085722/hfs-windows-x64-0.54.0-alpha1.3.zip)
